jax/numpy/lax_numpy.py

```diff
@@ -146,12 +146,12 @@ def _promote_dtypes(*args):
   if len(args) < 2:
     return args
   else:
-    from_dtypes = (_dtype(x) for x in args)
+    from_dtypes = (x if type(x) in (int, float, complex) else _dtype(x)
```

Comment:
You can gate on `six.PY2` to determine whether long exists.

Reply:
Done.